tf_encrypted/keras/layers/layers_utils.py

```diff
@@ -0,0 +1,25 @@
+"""TF Encrypted Keras layers utils"""
+import inspect
+
+import tensorflow as tf
+
+
+def not_implemented_arg_err(arg, arg_name, tf_layer_name):
```

Comment:
I think this should be implemented as a subclass of `NotImplementedError` and renamed to something like `LayerArgNotImplementedError`.  The arguments to this function would become arguments to the subclass's constructor, and you can override the `__str__` method of the subclass to return a custom error message.  The logic of this function can be implemented in the constructor or in the `__str__` method, whichever seems best.

Reply:
Sounds good, let's do that.

Reply:
I have subclassed `NotImplementedError`. However not sure how the following logic could be included in the class and how `LayerArgNotImplementedError` would be called in each layer:

```
tf_layer_cls = getattr(tf.keras.layers, tf_layer_name)
layer_sign = inspect.signature(tf_layer_cls.__init__).parameters
 tf_default_arg = layer_sign[arg_name].default
 if arg != tf_default_arg:
   raise LayerArgNotImplementedError(arg_name, tf_layer_name, tf_default_arg)
```

That's why I kept for now `not_implemented_arg_err`.